scripts/crusher_1D.py

```diff
@@ -193,20 +193,20 @@ def eval(self, values, x):
 F_t_rho = (sol_rho - sol_rho_h) * del_rho * dx
 F_t_J = (sol_J - sol_J_h) * del_J * dx
 
-F_rho_1 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0), del_rho, u_0, Nu_h_rho)) * dx
-F_J_1 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1), del_J, u_0, Nu_h_J, chi_h)) * dx
+F_rho_1 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0), del_rho, u, Nu_h_rho)) * dx
+F_J_1 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1), del_J, u, Nu_h_J, chi_h)) * dx
 F_1 = F_rho_1 + F_J_1
 
-F_rho_2 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0) + .5 * dT * kh_rho[0], del_rho, u_0, Nu_h_rho)) * dx
-F_J_2 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1) + .5 * dT * kh_J[0], del_J, u_0, Nu_h_J, chi_h)) * dx
+F_rho_2 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0) + .5 * dT * kh_rho[0], del_rho, u, Nu_h_rho)) * dx
+F_J_2 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1) + .5 * dT * kh_J[0], del_J, u, Nu_h_J, chi_h)) * dx
 F_2 = F_rho_2 + F_J_2
 
-F_rho_3 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0) + .5 * dT * kh_rho[1], del_rho, u_0, Nu_h_rho)) * dx
-F_J_3 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1) + .5 * dT * kh_J[1], del_J, u_0, Nu_h_J, chi_h)) * dx
+F_rho_3 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0) + .5 * dT * kh_rho[1], del_rho, u, Nu_h_rho)) * dx
+F_J_3 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1) + .5 * dT * kh_J[1], del_J, u, Nu_h_J, chi_h)) * dx
 F_3 = F_rho_3 + F_J_3
 
-F_rho_4 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0) + dT * kh_rho[2], del_rho, u_0, Nu_h_rho)) * dx
-F_J_4 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1) + dT * kh_J[2], del_J, u_0, Nu_h_J, chi_h)) * dx
+F_rho_4 = (ki_rho * del_rho + F_spatial_visc(sol_h.sub(0) + dT * kh_rho[2], del_rho, u, Nu_h_rho)) * dx
+F_J_4 = (ki_J * del_J + F_spatial_visc_chi(sol_h.sub(1) + dT * kh_J[2], del_J, u, Nu_h_J, chi_h)) * dx
 F_4 = F_rho_4 + F_J_4
```

Comment:
I guess that you are solving for the density `rho` and the scalar microinertia `J`, right?